plugin/sycl/objective/multiclass_obj.cc

```diff
@@ -0,0 +1,258 @@
+/*!
+ * Copyright 2015-2023 by Contributors
+ * \file multiclass_obj.cc
+ * \brief Definition of multi-class classification objectives.
+ */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtautological-constant-compare"
+#pragma GCC diagnostic ignored "-W#pragma-messages"
+#include <rabit/rabit.h>
+#pragma GCC diagnostic pop
+
+#include <vector>
+#include <algorithm>
+#include <limits>
+#include <utility>
+
+#include "xgboost/parameter.h"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtautological-constant-compare"
+#include "xgboost/data.h"
+#pragma GCC diagnostic pop
+#include "xgboost/logging.h"
+#include "xgboost/objective.h"
+#include "xgboost/json.h"
+
+#include "../device_manager.h"
+#include <CL/sycl.hpp>
+
+namespace xgboost {
+namespace sycl {
+namespace obj {
+
+DMLC_REGISTRY_FILE_TAG(multiclass_obj_sycl);
+
+/*!
+ * \brief Do inplace softmax transformaton on start to end
+ *
+ * \tparam Iterator Input iterator type
+ *
+ * \param start Start iterator of input
+ * \param end end iterator of input
+ */
+template <typename Iterator>
+inline void Softmax(Iterator start, Iterator end) {
+  bst_float wmax = *start;
+  for (Iterator i = start+1; i != end; ++i) {
+    wmax = ::sycl::max(*i, wmax);
+  }
+  float wsum = 0.0f;
+  for (Iterator i = start; i != end; ++i) {
+    *i = ::sycl::exp(*i - wmax);
+    wsum += *i;
+  }
+  for (Iterator i = start; i != end; ++i) {
+    *i /= static_cast<float>(wsum);
+  }
+}
+
+/*!
+ * \brief Find the maximum iterator within the iterators
+ * \param begin The begining iterator.
+ * \param end The end iterator.
+ * \return the iterator point to the maximum value.
+ * \tparam Iterator The type of the iterator.
+ */
+template<typename Iterator>
+inline Iterator FindMaxIndex(Iterator begin, Iterator end) {
+  Iterator maxit = begin;
+  for (Iterator it = begin; it != end; ++it) {
+    if (*it > *maxit) maxit = it;
+  }
+  return maxit;
+}
```

Comment:
Any chance we can use the one in `common/math.h`?

Reply:
done